### PR TITLE
INTCMN-502 - Do not perform user/group check before adding user/group

### PR DESCRIPTION
**Pull request by DMaxfield-BDS**

### CODE REVIEW

**Review by jamesrichard91 — APPROVED** [review]

File: src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java [hunk 1]
```diff
@@ -30,123 +34,158 @@
 public class UserServiceTestIT {
     private static final IntHttpClientTestHelper INT_HTTP_CLIENT_TEST_HELPER = new IntHttpClientTestHelper();
 
+    private final BlackDuckServicesFactory BLACKDUCK_SERVICES_FACTORY = UserServiceTestIT.INT_HTTP_CLIENT_TEST_HELPER.createBlackDuckServicesFactory();
```
Comment (ekerwin) [inline src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java:37]: the ALL_CAPS thing is usually reserved for static constants. I'd suggest making these static, or if that's not feasible, default to camelCase.

File: src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java [hunk 2]
```diff
@@ -30,123 +34,158 @@
...
-        UserGroupService userGroupService = blackDuckServicesFactory.createUserGroupService();
-        ProjectUsersService projectUsersService = blackDuckServicesFactory.createProjectUsersService();
-
         String projectName = "user-group-project" + System.currentTimeMillis();
         String userGroupName = "user-group-test" + System.currentTimeMillis();
         ProjectView projectView = null;
         UserGroupView userGroupView = null;
 
         try {
             ProjectSyncModel projectSyncModel = ProjectSyncModel.createWithDefaults(projectName, "test");
```
Comment (ekerwin) [inline src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java:65]: going back to camelCase would remove a lot of these diffs...not a huge plus, but perhaps worth considering

File: src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java [hunk 3]
```diff
@@ -30,123 +34,158 @@
...
+        userRequest.setLastName("IntegrationTest");
+        userRequest.setActive(active);
+        userRequest.setEmail("noreply@synopsys.com");
+        userRequest.setPassword("53CUR17y7hR0ugH085Cur17y");
+
+        HttpUrl userUrl = BLACKDUCK_API_CLIENT.post(ApiDiscovery.USERS_LINK, userRequest);
+        return BLACKDUCK_API_CLIENT.getResponse(userUrl, UserView.class);
+    }
+
+    private void cleanView(BlackDuckView blackDuckView) {
```
Comment (ekerwin) [inline src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java:183]: why cleanView over deleteView?

File: src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java [hunk 4]
```diff
@@ -30,123 +34,158 @@
...
+        }
+    }
+
+    @Test
+    public void testGetActiveUsersForProject() throws IntegrationException {
+        String projectName = "user-group-project" + System.currentTimeMillis();
+        String inActiveUser = "inactive-user" + System.currentTimeMillis();
+        String activeUser = "active-user" + System.currentTimeMillis();
+        ProjectView projectView = null;
+        UserView inActiveUserView = null;
```
Comment (ekerwin) [inline src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java:141]: I think 'inactive' is one word, so here, it seems like you are trying to declare a boolean flag as to whether or not you are in or out of 'active user view'

File: src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java [hunk 5]
```diff
@@ -30,123 +34,158 @@
...
+            assertFalse(allActiveUsersForProject.contains(inActiveUserView));
+            assertFalse(allActiveUsersForProject.contains(activeUserView));
+
+            PROJECT_USERS_SERVICE.addUserToProject(projectView, inActiveUser);
+            PROJECT_USERS_SERVICE.addUserToProject(projectView, activeUser);
+
+            allActiveUsersForProject = PROJECT_USERS_SERVICE.getAllActiveUsersForProject(projectView);
+            assertFalse(allActiveUsersForProject.contains(inActiveUserView));
+            assertTrue(allActiveUsersForProject.contains(activeUserView));
+        } finally {
```
Comment (ekerwin) [inline src/test/java/com/synopsys/integration/blackduck/service/dataservice/UserServiceTestIT.java:163]: Given that we've removed the check for existing users, we could use this current stone to kill a few more birds and try adding the users AGAIN and verify that there are no exceptions and the returned list doesn't change in terms of size or content.

**Pull request merged**